cli/commands/terraform/target.go

```diff
@@ -0,0 +1,39 @@
+package terraform
+
+import (
+	"github.com/gruntwork-io/terragrunt/config"
+	"github.com/gruntwork-io/terragrunt/options"
+)
+
+const (
+	TargetPointParseConfig TargetPointType = iota + 1
+	TargetPointDownloadSource
+	TargetPointGenerateConfig
+	TargetPointInitCommand
+)
+
+type TargetPointType byte
+
+type TargetCallbackType func(opts *options.TerragruntOptions, config *config.TerragruntConfig) error
+
+// Since most terragrunt commands like `render-json`, `aws-provider-patch` ...  require preparatory steps.
+// Target helps to execute a command up to a certain point and then run a callback function.
```

Comment:
I'm not sure I understand this or what `Target` represents. Same for `isPoint` or the need for callbacks. 

Can you provide a bit more context behind this structure?

Reply:
Got it, thanks for the explanation. I agree that `runTerraform` is way too big and complicated and needs to be simplified. I'm a bit worried that the target/isPoint approach actually makes it _more_ complicated. But if it's an incremental stepping stone—a way to avoid making this PR even bigger and take even longer—I'm OK with that, as long as the next step is to refactor it into something simpler / more modular, and remove the target/isPoint behavior.

Reply:
Totally agree with everything you said. The next step will be refactoring the `runTerraform` function.